CMakeLists.txt

```diff
@@ -66,7 +66,7 @@ IF (NOT KOKKOSKERNELS_HAS_TRILINOS)
           "ENABLE_PERFTESTS"
           ON
           BOOL
-          "Whether to build performance tests. Default: OFF"
+          "Whether to build performance tests. Default: ON"
```

Comment:
We should turn these OFF by default since that's what we have for tests and examples.
I would leave the doc string as is but change the value on line 67

Reply:
FWIW, cm_test_all_sandia and cm_generate_makefile.bash defaults perftests to ON.

Reply:
@lucbv I like that too actually. I just made the default OFF. Now that both perftests and examples are OFF by default, I changed the messages for when they're ON but ``ENABLE_ALL_COMPONENTS`` is OFF a warning.

Before, I didn't want a build with some components disabled (but otherwise a default build) throwing warnings.